README.md
# 2bij3

News
Configured with email username  None , password  False
_sa_instance_state, elasticsearch, id, intensity, negativity, position, recommended, timestamp, url, user_id
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186670>","38","1","None","None","0","1","2021-07-28 13:45:30.949526","https://krant.ad.nl/titles/algemeendagblad/7908/publications/25349/articles/1416431/17/1","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186760>","16","2","None","None","1","1","2021-07-28 13:45:33.030031","https://www.nrc.nl/nieuws/2021/07/27/canadese-eigenaren-zetten-de-bijenkorf-te-koop-a4052613#20210728-79721-301","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b71867f0>","69","3","None","None","2","1","2021-07-28 13:45:33.047439","https://digitalekrant.telegraaf.nl/static/cci/index.html?epub=https://digitalekrant.telegraaf.nl/cdn/premium/45cb788b2c96c2d824181a3b4d5c9351/web/OPS/cciobjects.json&hash=59a5843373d7dbecf845#/pages/28","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186880>","10","4","None","None","3","1","2021-07-28 13:45:33.665852","https://krant.ad.nl/titles/algemeendagblad/7908/publications/25349/articles/1416431/3/1","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186940>","56","5","None","None","4","1","2021-07-28 13:45:34.550370","https://krant.trouw.nl/titles/trouw/8321/publications/1313/articles/1416415/11/2","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186a00>","25","6","None","None","5","1","2021-07-28 13:45:35.140585","https://krant.volkskrant.nl/titles/volkskrant/7929/publications/1318/articles/1416433/10/1","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186ac0>","21","7","None","None","0","1","2021-07-28 13:47:27.741721","https://digitalekrant.telegraaf.nl/static/cci/index.html?epub=https://digitalekrant.telegraaf.nl/cdn/premium/45cb788b2c96c2d824181a3b4d5c9351/web/OPS/cciobjects.json&hash=2bbfbd19fdeff5577be1#/pages/4-5","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186b80>","15","8","None","None","1","1","2021-07-28 13:47:28.348696","https://www.nrc.nl/nieuws/2021/07/27/canadese-eigenaren-zetten-de-bijenkorf-te-koop-a4052613#20210728-79721-301","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186c40>","7","9","None","None","2","1","2021-07-28 13:47:29.053708","https://digitalekrant.telegraaf.nl/static/cci/index.html?epub=https://digitalekrant.telegraaf.nl/cdn/premium/45cb788b2c96c2d824181a3b4d5c9351/web/OPS/cciobjects.json&hash=ef992ebe7605dbc62be7#/pages/20-21","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186d00>","67","10","None","None","3","1","2021-07-28 13:47:29.089041","https://digitalekrant.telegraaf.nl/static/cci/index.html?epub=https://digitalekrant.telegraaf.nl/cdn/premium/45cb788b2c96c2d824181a3b4d5c9351/web/OPS/cciobjects.json&hash=4d559e04f8ddf3775df5#/pages/28","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186dc0>","42","11","None","None","4","1","2021-07-28 13:47:29.698879","https://krant.trouw.nl/titles/trouw/8321/publications/1313/articles/1416415/15/2","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186e80>","50","12","None","None","5","1","2021-07-28 13:47:30.010950","https://digitalekrant.telegraaf.nl/static/cci/index.html?epub=https://digitalekrant.telegraaf.nl/cdn/premium/45cb788b2c96c2d824181a3b4d5c9351/web/OPS/cciobjects.json&hash=d90e0752ad6fc0b298ff#/pages/38","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b7186f40>","64","13","None","None","0","1","2021-07-28 13:50:16.256652","https://krant.volkskrant.nl/titles/volkskrant/7929/publications/1318/articles/1416433/21/1","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b6a89040>","56","14","None","None","1","1","2021-07-28 13:50:16.599490","https://krant.trouw.nl/titles/trouw/8321/publications/1313/articles/1416415/11/2","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b6a89100>","62","15","None","None","2","1","2021-07-28 13:50:16.660134","https://krant.trouw.nl/titles/trouw/8321/publications/1313/articles/1416415/14/2","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b6a891c0>","9","16","None","None","3","1","2021-07-28 13:50:16.678073","https://krant.volkskrant.nl/titles/volkskrant/7929/publications/1318/articles/1416433/11/2","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b6a89280>","10","17","None","None","4","1","2021-07-28 13:50:17.107724","https://krant.ad.nl/titles/algemeendagblad/7908/publications/25349/articles/1416431/3/1","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b6a89340>","25","18","None","None","5","1","2021-07-28 13:50:17.905725","https://krant.volkskrant.nl/titles/volkskrant/7929/publications/1318/articles/1416433/10/1","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b6a89400>","33","19","None","None","0","1","2021-07-28 13:51:01.088105","https://digitalekrant.telegraaf.nl/static/cci/index.html?epub=https://digitalekrant.telegraaf.nl/cdn/premium/45cb788b2c96c2d824181a3b4d5c9351/web/OPS/cciobjects.json&hash=69cfa21acd261a6aac1b#/pages/8","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b6a894c0>","31","20","None","None","1","1","2021-07-28 13:51:01.299174","https://digitalekrant.telegraaf.nl/static/cci/index.html?epub=https://digitalekrant.telegraaf.nl/cdn/premium/45cb788b2c96c2d824181a3b4d5c9351/web/OPS/cciobjects.json&hash=3a7bd96d1009304d5f11#/pages/28","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b6a89580>","39","21","None","None","2","1","2021-07-28 13:51:01.602814","https://digitalekrant.telegraaf.nl/static/cci/index.html?epub=https://digitalekrant.telegraaf.nl/cdn/premium/45cb788b2c96c2d824181a3b4d5c9351/web/OPS/cciobjects.json&hash=710e8c9c0922b1f189e3#/pages/20-21","1",
"<sqlalchemy.orm.state.InstanceState object at 0x7f49b6a89640>","49","22","None","None","3","1","2021-07-28 13:51:03.189385","https://krant.ad.nl/titles/algemeendagblad/7908/publications/25349/articles/1416431/6/1","1",

app/vars.py
'''
In this file you can specify all the variables
to customize the application.
'''

'''
ELASTICSEARCH FIELDS (examples given are from INCA database)

textfield is the elasticsearch field where the text can be found (should already be processed and tokenized)
teaserfield is the elasticsearch field where the teaser can be found
teaseralt is the elasticsearch field where an alternative teaser can be found (e.g. a rss teaser)
doctypefield is the elasticsearch field where the document type is stored
classifier_dict is the dictionary that has the results of a prediction with the topic classifier as key and the topic or topics (as list) as value
'''
topicfield = "section"
textfield = "text"
teaserfield = ""
teaseralt = ""
doctypefield = "publisher"
titlefield = "title"
'''
ELASTICSEARCH SETUP (examples given are from INCA database)
'''
from elasticsearch import Elasticsearch
host = "http://localhost:9200"
indexName = "inca"
es = Elasticsearch(host, timeout = 60)

'''
DOCUMENT TYPES
list_of sources: Which document types do you want to use? They should be in the doctypefield you supplied
doctype_dict: How the sources will be displayed to the user
topics: whether a topic variable will be used/displayed to the user
'''
list_of_sources = ["nu", "bd (www)", "telegraaf (www)", "volkskrant (www)", "ad (www)"]
doctype_dict = {'telegraaf (www)': 'telegraaf.nl', 'nu': 'nu.nl', 'bd (www)':'bd.nl', 'volkskrant (www)': 'volkskrant.nl', 'ad (www)':'ad.nl'}
topics = True

'''
NUMBER OF ARTICLES AND GROUPS
num_less is the initial number of articles per source that will be scraped,
num_more is the number that will be used when running out of stories(e.g. person has already seen all the stories retrieved)
num_select is the number of stories that will be displayed to the user
num_recommender is the number of stories that will be chosen by the recommender (if applicable)
group_number is how many different groups (i.e. recommenders) do you have?
'''
num_less = 20
num_more = 200
num_select = 6
num_recommender = 1
group_number = 1

'''
REQUIREMENTS FOR FINISHING STUDY
p1_day_min: How many days need participants to use the application in phase 1
p1_points_min: How many points do they need to collect in phase 1
p2_day_min: How many days need participants to use the application in phase 1 and 2
p2_points_min: How many points do they need to collect in phase 1 and 2
'''
p1_day_min = 2
p1_points_min = 14
p2_day_min = 2
p2_points_min = 14
'''
TOPICS
topic_list: The different topic categories that can be displayed to the user
classifier_dict: map numbers in elasticsearch database to strings (for the topic tag)
all_categories: map topic strings to topic numbers (that are stored in the SQL database)
'''
topic_list = ["Binnenland","Buitenland", "Economie", "Milieu", "Wetenschap", "Immigratie",\
"Justitie","Sport","Entertainment","Anders"]

classifier_dict = {topic_list[0]:['13','14','20', '3', '4', '5', '6'], topic_list[1]:['16', '19', '2'],\
 topic_list[2]:['1','15'], topic_list[3]:['8', '7'],  topic_list[4]:['17'], topic_list[5]:['9'],  topic_list[6]:['12'],\
  topic_list[7]:['29'], topic_list[8]:['23'], topic_list[9]:['10','99']}

all_categories = {"topic1":topic_list[0], "topic2":topic_list[1], "topic3":topic_list[2], "topic4":topic_list[3],\
 "topic5":topic_list[4], "topic6":topic_list[5], "topic7":topic_list[6], "topic8":topic_list[7], \
 "topic9":topic_list[8], "topic10":topic_list[9]}
